CAP/gap/DerivedMethods.gi

```diff
@@ -663,45 +663,65 @@ AddDerivationToCAP( UniversalMorphismFromCoequalizer,
 ##
 AddDerivationToCAP( UniversalMorphismIntoFiberProduct,
```

Comment:
You should be able to simply remove those derivations. They should be covered by the derivation of `FiberProduct` from `DirectProduct` and `Equalizer`.

Reply:
I was so focused on replacing, that I didn't think about it. Thanks!

Is this also true for `FiberProductEmbeddingInDirectSum` and `DirectSumProjectionInPushout`? If yes, then it's not clear to me.

Reply:
> General answer: If `FiberProductEmbeddingInDirectSum` is more or less only used to derive `FiberProduct` (just like `DirectSumDiagonalDifference`) it should also be deleted. If it has actual applications, it should be generalized to `FiberProductEmbeddingInDirectProduct`

I could only find this instance, which looks like an application:

https://github.com/homalg-project/CAP_project/blob/96d770cff4bbd7768c72444d8dc21d81140b0102/GeneralizedMorphismsForCAP/examples/Sweep.g#L15-L16

And this, which I'm not sure about:

https://github.com/homalg-project/CAP_project/blob/96d770cff4bbd7768c72444d8dc21d81140b0102/AttributeCategoryForCAP/gap/AttributeCategory.gi#L490-L502

> Concrete answer: I cannot really tell without getting an overview over how `FiberProductEmbeddingInDirectSum` is used. If you think it's clear, feel free to vote for one solution or the other :-)

I would say, we postpone this until Thursday, where I would show it to @mohamed-barakat.
But either way, if `FiberProductEmbeddingInDirectSum` needs to use `DirectProduct` or can be deleted, I would do the changes in a separate PR, since both solutions are bigger changes. 

And I will remove the derivation of `UniversalMorphismIntoFiberProduct` together with the remnants of `DirectSumDiagonalDifference` in the next PR.

So I would say, this is ready for review.